src/client/public/scripts/app/router.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * This adds a global event listener for URL changes and fires a 'url-changed'
+ * event with the new path upon same-origin URL navigations.
+ */
+
+/**
+ * Fire an event if there are same origin changes.
+ */
+function globalClickHandler(event: MouseEvent) {
+  if (event.defaultPrevented) {
+    return;
+  }
+
+  const path = getSameOriginLink(event);
```

Comment:
Maybe rename this to isSameOriginLink or filterSameOriginLink.

Took me a while to figure out what this was for.

Reply:
Done